src/compiler/checker.ts

```diff
@@ -3790,6 +3796,28 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return false;
     }
 
+    function isExtendedByClass(node: Node): boolean {

```

Comment:
Instead of creating two new functions, can you instead rename `isExtendedByInterface` to `isInheritedByClassOrInterface` and add parameters `heritageKind: SyntaxKind, containerKind: SyntaxKind` for calculating `isExtending` and `isInterface`, respectively?

Reply:
I agree that the `isExtendedByInterface`, `isExtendedByClass`, and `isImplementedByClass` functions are similar. I tried to merge them into a single function called `isInheritedByClassOrInterface`, but realized it is redundant because it simply returns true or false after checking if the node contains a HeritageClause and a Class or Interface. I couldn't decide which error message to print with single boolean value returned by `isInheritedByClassOrInterface`.

Therefore, I think it's best to move all of the logic for checking these conditions into the `checkAndReportErrorForUsingTypeAsValue` function.
You can find changes in a next commit.